includes/wikia/memcached/MemcachedStats.class.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * Memcache stats collector
+ *
+ * @author macbre
+ */
+namespace Wikia\Memcached;
+
+use Wikia\Util\Statistics\BernoulliTrial;
+
+class MemcachedStats {
+
+	/**
+	 * Get memcache stats:
+	 *  - top keys (both hits and misses)
+	 *  - memcache operations counter (get, set, miss, delete, add)
+	 *
+	 * @return array
+	 */
+	private static function getStats() {
+		/* @var $wgMemc \MemcachedPhpBagOStuff */
+		global $wgMemc;
+		$client = $wgMemc->getClient();
+
+		$stats = [
+			'counts' => $client->stats,
+			'keys' => [
+				'hits' => self::normalizeKeys($client->keys_stats['hits']),
+				'misses' => self::normalizeKeys($client->keys_stats['misses']),
+			]
+		];
+
+		return $stats;
+	}
+
+	/**
+	 * Normalizes the given set of keys
+	 *
+	 * @param array $keys
+	 * @return array normalized keys
+	 */
+	private static function normalizeKeys(Array $keys) {
+		return array_map([__CLASS__, 'normalizeKey'], $keys);
+	}
+
+	/**
+	 * Normalizes given key by removing the wiki-specific prefix and
+	 * removing hashes and IDs parts from the key
+	 *
+	 * @param string $key
+	 * @return string
+	 */
+	public static function normalizeKey($key) {
+		// remove per-wiki prefix
+		$prefix = \F::app()->wf->WikiId() . ':';
+		if (startsWith($key, $prefix)) {
+			$key = str_replace($prefix, '*:', $key);
+		}
+
+		$parts = array_map(
```

Comment:
Generating stats for around 100 keys takes 0,2 - 0,3 ms. I don't think it's worth the optimization.